src/table.js

```diff
@@ -314,6 +317,184 @@ export const addColumnAt = columnIndex => tr => {
+// Returns a new transaction that moves the origin row to the target index;
+// this does not affect the current table; it returns a new copy with a further table.
+//
+// for default "tryToFit" is false, that's means if you try to move a row to a place
+// where we will need to split a row with cell merges it'll blow an exception, for example:
+//
+//      ____________________________
+//     |      |      |             |
+//  0  |  A1  |  B1  |     C1      |
+//     |______|______|______ ______|
+//     |      |             |      |
+//  1  |  A2  |     B2      |      |
+//     |______|______ ______|      |
+//     |      |      |      |  D1  |
+//  2  |  A3  |  B3  |  C2  |      |
+//     |______|______|______|______|
+//
+// if you try to move the row 0 to the position 1 with tryToFit false,
+// it'll throw an exception since you can't split the row 1;
+// but if "tryToFit" is true, it'll move the row using the current direction.
+//
+// We defined current direction using the target and origin values
+// if the origin is greater than the target, that's means the course is `right-to-left`,
+// so the `tryToFit` logic will use this direction to determinate
+// if we should move the column to the right or the left.
+//
+// for example, if you call the function using `moveRow(0, 1, { tryToFit: true })`
+// the result will be:
+//      ____________________________
+//     |      |             |      |
+//  0  |  A2  |     B2      |      |
+//     |______|______ ______|      |
+//     |      |      |      |  D1  |
+//  1  |  A3  |  B3  |  C2  |      |
+//     |______|______|______|______|
+//     |      |      |             |
+//  2  |  A1  |  B1  |     C1      |
+//     |______|______|______ ______|
+//
+// since we could put the row zero on index one,
+// we pushed to the best place to fit the column 0,
+// in this case, position 2.
+//
+//
+// ```javascript
+// dispatch(
+//   moveRow(x, y, options)(state.tr)
+// );
+// ```
+export const moveRow = (originRowIndex, targetRowIndex, opts) => tr => {
+  const defaultOptions = { tryToFit: false, direction: 0 };
+  const options = Object.assign(defaultOptions, opts);
+  const table = findTable(tr.selection);
+  if (!table) {
+    return tr;
+  }
+
+  const { indexes: indexesOriginRow } = getSelectionRangeInRow(originRowIndex)(
+    tr
+  );
+  const { indexes: indexesTargetRow } = getSelectionRangeInRow(targetRowIndex)(
+    tr
+  );
+
+  if (indexesOriginRow.indexOf(targetRowIndex) > -1) {
+    return tr;
+  }
+
+  if (!options.tryToFit && indexesTargetRow.length > 1) {
+    checkInvalidMovements(originRowIndex, targetRowIndex, indexesTargetRow);
+  }
+
+  const newTable = moveTableRow(table, indexesOriginRow, indexesTargetRow);
+
+  return cloneTr(tr).replaceWith(
+    table.pos,
+    table.pos + table.node.nodeSize,
+    newTable
+  );
+};
+
+// :: (originColumnIndex: number, targetColumnIndex: targetColumnIndex, options: MovementOptions) → (tr: Transaction) → Transaction
```

Comment:
* should `options ` be an optional argument? right now it seems like its required
* where `MovementOptions` are defined?
* should we pass all arguments as an object instead? e.g. `moveColumn({ from: 0, to: 2, tryToFit: true })`

Reply:
1) Fixed
2) It's defined on `typing.ts`, It was called `MoveOptions` but I renamed it to `MovementOptions`
3) No idea, I'm okay with anyone